### PR TITLE
[New Rule] Persistence via KDE AutoStart Script or Desktop File Modif…

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/808

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/linux/persistence_kde_autostart_modification.toml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+         "/home/*/.config/autostart/*", 
+         "/home/*/.kde/Autostart/*", 
+         "/home/*/.kde4/Autostart/*", 
+         "/home/*/.kde/share/autostart/*", 
+         "/home/*/.kde4/share/autostart/*", 
+         "/home/*/.local/share/autostart/*", 
+         "/home/*/.config/autostart-scripts/*", 
+         "/etc/xdg/autostart/*", 
+         "/usr/share/autostart/*"
+        )
```
Comment (bm11100) [inline rules/linux/persistence_kde_autostart_modification.toml:42]:
```suggestion
file where event.type != "deletion" and
  file.extension in ("sh", "desktop") and
  file.path :
    (
      "/home/*/.config/autostart/*",
      "/home/*/.kde/Autostart/*",
      "/home/*/.kde4/Autostart/*", 
      "/home/*/.kde/share/autostart/*",
      "/home/*/.kde4/share/autostart/*",
      "/home/*/.local/share/autostart/*",
      "/home/*/.config/autostart-scripts/*",
      "/etc/xdg/autostart/*",
      "/usr/share/autostart/*"
    )
```

File: rules/linux/persistence_kde_autostart_modification.toml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+[metadata]
+creation_date = "2020/01/06"
+maturity = "production"
+updated_date = "2020/01/06"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies modification or creation of a KDE AutoStart script or desktop file that will execute at every user logon.
+Adversaries may use this method for persistence.
```
Comment (bm11100) [inline rules/linux/persistence_kde_autostart_modification.toml:10]:
```suggestion
Identifies the creation or modification of a K Desktop Environment (KDE) AutoStart script or desktop file that will execute upon each user logon. Adversaries may abuse this method for persistence.
```

File: rules/linux/persistence_kde_autostart_modification.toml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+[metadata]
+creation_date = "2020/01/06"
+maturity = "production"
+updated_date = "2020/01/06"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation or modification of a K Desktop Environment (KDE) AutoStart script or desktop file that will execute upon each user logon. Adversaries may abuse this method for persistence.
```
Comment (bm11100) [inline rules/linux/persistence_kde_autostart_modification.toml:9]: just a re-lint, otherwise LGTM

**Review by bm11100 — APPROVED** [review]

File: rules/linux/persistence_kde_autostart_modification.toml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  file.extension in ("sh", "desktop") and
+  file.path :
+    (
+      "/home/*/.config/autostart/*",
+      "/home/*/.kde/Autostart/*",
+      "/home/*/.kde4/Autostart/*", 
+      "/home/*/.kde/share/autostart/*",
+      "/home/*/.kde4/share/autostart/*",
+      "/home/*/.local/share/autostart/*",
+      "/home/*/.config/autostart-scripts/*",
```
Comment (brokensound77) [inline rules/linux/persistence_kde_autostart_modification.toml:39]: Will the same behavior apply to the root user too?

```suggestion
      "/home/*/.config/autostart/*", "/root/.config/autostart/*",
      "/home/*/.kde/Autostart/*", "/root/.kde/Autostart/*",
      "/home/*/.kde4/Autostart/*", "/root/.kde4/Autostart/*",
      "/home/*/.kde/share/autostart/*", "/root/.kde/share/autostart/*",
      "/home/*/.kde4/share/autostart/*", "/root/.kde4/share/autostart/*",
      "/home/*/.local/share/autostart/*", "/root/.local/share/autostart/*",
      "/home/*/.config/autostart-scripts/*", "/root/.config/autostart-scripts/*",
```

Comment (Samirbous) [reply rules/linux/persistence_kde_autostart_modification.toml:39]: guess yes, good catch!

**Review by threat-punter — APPROVED** [review]

**Pull request merged**